devito/dimension.py

```diff
@@ -0,0 +1,22 @@
+from sympy import Symbol
+
+__all__ = ['Dimension', 'x', 'y', 'z', 't']
+
+
+class Dimension(Symbol):
```

Comment:
Just to be picky: how about ``LoopIndex`` or ``IndexSymbol`` or ``LoopDimension`` ? 

Because I know understand what a ``Dimension`` is, but first time I read the name it felt weird for a "Dimension" to be a "Symbol"